### PR TITLE
CE-3301 Modify RecentChanges API to return a URL to a user avatar

### PR DESCRIPTION
**Pull request by adamkarminski**
@Wikia/community-engineering

### CODE REVIEW

File: includes/api/ApiQueryRecentChanges.php [hunk 1]
```diff
@@ -408,6 +410,10 @@ public function extractRowInfo( $row ) {
 				$vals['userid'] = $row->rc_user;
 			}
 
+			if ( $this->fld_useravatar ) {
+				$vals['useravatar'] = AvatarService::getAvatarUrl( $vals['user'], AvatarService::AVATAR_SIZE_MEDIUM );
```
Comment (Grunny) [inline includes/api/ApiQueryRecentChanges.php]: If someone sets the props as `userid|useravatar`, this code would be executed but `$vals['user']` would be undefined as the `user` prop wasn't set, so it's probably better to use `$row->rc_user_text` here.

Comment (adamkarminski) [inline includes/api/ApiQueryRecentChanges.php]: Done!

**Comment by lukaszkonieczny** [comment]
LGTM

File: includes/api/ApiQueryRecentChanges.php [hunk 2]
```diff
@@ -639,6 +646,7 @@ public function getParamDescription() {
 				'Include additional pieces of information',
 				' user           - Adds the user responsible for the edit and tags if they are an IP',
 				' userid         - Adds the user id responsible for the edit',
+				' useravatar     - Adds an avatar of a user responsible for the edit. It requires the user or the userid param to be present.',
```
Comment (kamilkoterba) [inline includes/api/ApiQueryRecentChanges.php:649]: :+1:

**Comment by kamilkoterba** [comment]
LGTM

**Comment by azarek** [comment]
LGTM

**Pull request merged**